src/command_modules/azure-cli-cosmosdb/azure/cli/command_modules/cosmosdb/custom.py

```diff
@@ -226,6 +227,10 @@ def _populate_collection_definition(collection,
         collection['partitionKey'] = {'paths': [partition_key_path]}
         changed = True
 
+    if defaultTtl:
+        changed = True
```

Comment:
It seems you could simplify this to something like:
```Python
if not any(partition_key_path, default_ttl, ...):
  return False

if default_ttl:
  collection['defaultTtl'] = default_ttl

...

return True
```

Then you don't need all the "changed = True" statements.